File: rust/leetcode_rust/src/leetcode/richest_customer_wealth_1672.rs
pub(crate) struct RichestCustomerWealth {
    accounts: [[i32; 3]; 3], // like self.accounts
}

impl RichestCustomerWealth {
    // Constructor method
    pub(crate) fn new(accounts: [[i32; 3]; 3]) -> Self {
        RichestCustomerWealth { accounts } // value goes to self.accounts
    }

    pub fn calculate_richest_customer_wealth(&self) -> i32 {
        let mut highest_balance = 0;

        // Iterate over each account
        for row in self.accounts.iter() {
            // Sum the wealth of the current account
            let sum: i32 = row.iter().sum();

            // Update with highest_balance or sum, whichever is greater
            highest_balance = highest_balance.max(sum);  //
        }

        highest_balance  // Return the highest balance
    }
}

File: kotlin/leetcode_koltin/src/RichestCustomerWealth1672.kt
// Kotlin classes take constructor params in class signature
class RichestCustomerWealth1672(private val accounts: List<List<Int>>) {
    private var highestBalance = 0

    // init is like constructor but values passed in above in signature
    init {
        // Sum each account and find the highest balance
        for (row in accounts) {
            val sum = row.sum()
            highestBalance = if (sum > highestBalance) sum else highestBalance
        }
    }
    fun calculateRichestCustomerWealth(): Int {
        return highestBalance
    }
}
